homeassistant/components/media_player/kodi.py

```diff
@@ -520,25 +564,31 @@ def supported_features(self):
         """Flag media player features that are supported."""
         supported_features = SUPPORT_KODI
 
-        if self._turn_off_action in TURN_OFF_ACTION:
+        if self._turn_on_action is not None:
+            supported_features |= SUPPORT_TURN_ON
+
+        if self._turn_off_action is not None:
             supported_features |= SUPPORT_TURN_OFF
 
         return supported_features
 
     @cmd
     @asyncio.coroutine
+    def async_turn_on(self):
+        """Execute turn_on_action to turn on media player."""
+        if self._turn_on_action is not None:
+            yield from self._turn_on_action.async_run()
+            yield from self.async_update_ha_state(force_refresh=True)
```

Comment:
Ok, I'll do it as well with the `async_on_quit` call for the `turn_off_action` script, ok?